zucchini-ui-frontend/src/failures/components/FailuresTable.jsx

```diff
@@ -0,0 +1,148 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Link} from 'react-router';
+import Table from 'react-bootstrap/lib/Table';
+import Label from 'react-bootstrap/lib/Label';
+import Modal from 'react-bootstrap/lib/Modal';
+import Status from '../../ui/components/Status';
+
+export default class FailuresTable extends React.Component {
+
+  render() {
+    const {failures} = this.props;
+    const rows = failures.failures.map(groupedFailures => {
+      const nbFailedScenarii = groupedFailures.failedScenarii.length;
+      return (
+        groupedFailures.failedScenarii.map((failedScenario, index) => {
+          return <FailuresTableRow key={index} failedScenario={failedScenario} isFirstFailure={index === 0 } nbFailedScenarii={nbFailedScenarii}/>
+        })
+      );
+    });
+
+    return (
+      <Table bordered striped hover style={{tableLayout: 'fixed'}}>
+        <thead>
+          <tr>
+            <th className='col-md-4'>Erreur</th>
+            <th className='col-md-6'>Scénario</th>
+            <th className='col-md-1'>Statut</th>
+            <th className='col-md-1'>Analysé</th>
+          </tr>
+        </thead>
+        <tbody>{rows}</tbody>
+      </Table>
+    );
+  }
+
+}
+
+FailuresTable.propTypes = {
+  testRunId: PropTypes.string.isRequired,
+  failures: PropTypes.object.isRequired,
+};
+
+class FailuresTableRow extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showErrorMsgDetails: false,
+    };
+  }
+
+  onShowErrorMsg = () => {
+    this.setState({
+      showErrorMsgDetails: true,
+    });
+  };
+
+  onHideErrorMsg = () => {
+    this.setState({
+      showErrorMsgDetails: false,
+    });
+  };
+
+  handleErrorMsgRow(rowSpan, msg, scenario) {
+
+    const cellStyle = {
+      overflow: 'hidden',
+      textOverflow: 'ellipsis'
+    };
+
+    return (
+      <td rowSpan={rowSpan} style={cellStyle}>{msg}&hellip;<br/>
+        <a onClick={this.onShowErrorMsg}><b>Détails</b></a>
+        <FailuresDetailsDialog errorMessage={scenario.errorMessage} show={this.state.showErrorMsgDetails} onClose={this.onHideErrorMsg}/>
+      </td>
+    );
+  }
+
+  render() {
+    const {failedScenario, isFirstFailure, nbFailedScenarii} = this.props;
+
+    const reviewedProps = {
+      bsStyle: failedScenario.reviewed ? 'success' : 'default',
```

Comment:
You can add ternary expression directly in JSX, you don't need an extra variable.